src/chunkers/hybrid_chunker.py

```diff
@@ -0,0 +1,240 @@
+from __future__ import annotations
+
+"""Hybrid chunker implementation using Docling's HybridChunker."""
+import logging
+
+import tempfile
+from pathlib import Path
+from typing import Any, Optional
+
+from langchain.schema import Document
+
+from ..constants import DEFAULT_ENCODING
+from .protocol import Chunker
+from .tokenizers import Tokenizer, TokenizerFactory, TokenizerType
+
+from docling.chunking import HybridChunker as DoclingHybridChunker
+from docling.datamodel.base_models import InputFormat
+from docling.document_converter import DocumentConverter
+
+
+class HybridChunker:
+    """Hybrid chunker using Docling's semantic + fixed-size chunking.
+
+    This chunker combines semantic chunking with token-based fixed-size
+    chunking, optimized for Gemini models with a 2000 token limit.
+    """
+
+    def __init__(
+        self,
+        max_tokens: int = 2000,
+        merge_peers: bool = False,
+        tokenizer: Optional[Tokenizer] = None,
+    ):
+        """Initialize the hybrid chunker.
+
+        Parameters
+        ----------
+        max_tokens
+            Maximum tokens per chunk (default: 2000).
+        merge_peers
+            Whether to merge peer chunks (default: False).
+        tokenizer
+            Optional tokenizer instance. If not provided, uses a simple approximation tokenizer.
+        """
+        self.logger = logging.getLogger()
```

Comment:
```suggestion
        self.logger = logging.getLogger("HYBRID_CHUNKER")
```
We could implement logger names to have more information when an error ocurrs

Reply:
I like. We can do another PR putting names in each get logger call.